# HiCzin
HiCzin: Normalizing metagenomic Hi-C data and detecting spurious contacts using zero-inflated negative binomial regression

## Source of biases in metagenomic Hi-C experiments
In addition to chromosomal contacts of interests, several other factors unrelated to chromosomal contacts can influence the number of Hi-C interactions between contigs. We refer to such factors as biases. Two kinds of biases with substantial influences on metagenomic Hi-C contact maps were reported: explicit biases and implicit biases. Explicit biases include three potential factors: the number of enzymatic restriction sites on contigs, the length and the coverage of contigs. Implicit biases contain unobserved interactions and spurious inter-species contacts. Unobserved interactions mean that some chimerical DNA fragments are missed due to the factors such as mappability of contigs and in vivo constraints on accessibility. Spurious inter-species contacts arise from the ligation of DNA fragments between closely related species. It is necessary to correct these biases before downstream analysis.

## HiCzin: Normalizing metagenomic Hi-C data and detecting spurious contacts
HiCzin is a normalization method designed for metagenomic Hi-C data. HiCzin is based on the zero-inflated negative binomial regression frameworks. We model the population of the intra-species contacts using the negative binomial distribution. HiCzin combines the counting distribution of the intra-species contacts with a mass distribution of unobserved contacts. The residues of the counting part serve as normalized contacts. HiCzin can also be used to detect and remove spurious contacts.


## Usage 
HiCzin is available through the Rscript HiCzin.R. 

### Typical Workflow
#### Contig Assembly
For the shotgun library, de novo metagenome assembly was produced by MEGAHIT with parameters ‘-min-contig-len 300 -k-min 21 -k-max 141 -k-step 12 -merge-level 20,0.95’ and contigs shorter than 1 kb were discarded.
#### Calculating the coverage of assembled contigs
Firstly, BBmap from the BBTools suite was applied to map the shotgun reads to the assembled contigs with parameters ‘bamscript=bs.sh; sh bs.sh’. The coverage of contigs was computed using MetaBAT v2.12.5 script: ‘jgi summarize bam contig depths’.
#### Alignment of Hi-C paired-end reads
Hi-C paired-end reads were mapped to assembled contigs using BWA-MEM with parameters ‘-5SP’. Then, samtools with parameters ‘view -F 0x904’ were applied on the resulting BAM files to remove unmapped reads (0x4) and supplementary (0x800) and secondary (0x100) alignments. Alignments with low quality ( <30 nucleotide match length or mapping score <30) were also filtered out.
#### Generating sample data of the intra-species contacts by TAXAassign
The taxonomic assignment of contigs was resolved using NCBI’s Taxonomy and its nt database by TAXAassign(v0.4) with parameters ‘-p -c 20 -r 10 -m 98 -q 98 -t 95 -a “60,70,80,95,95,98” -f’. Assignment results with ‘unclassified’ at the species level were discarded, and only deterministic results of taxonomic assignment at the species level were kept. Intra-species pairs were subsequently generated by pairwise combining contigs assigned to the same species, and corresponding contacts were treated as samples to fit the HiCzin model.

### Option
```bash
-i --input:     input raw metagenomic Hi-C contacts(csv file)
-s --sample:    input samples of the intra-species contacts 
-o --output:    output normalized Hi-C contacts that are regarded as valid contacts and retained by HiCzin
-d --discard:   output normalized Hi-C contacts that are regarded as spurious contacts and discarded
-f --info:      contig information file
-c --coverage:  contigs' converage file
-t --threshold: preselected threshold for discarding spurious contacts
-u --unlabeled: use ulabeled mode of HiCzin
-n --nb:        use classical negative binomial regression
```

## Input and Output Format






